packages/ui/components/overlays/src/OverlayController.js

```diff
@@ -1126,9 +1128,14 @@ export class OverlayController extends EventTarget {
     }
   }
 
-  /** @private */
-  __escKeyHandler(/** @type {KeyboardEvent} */ ev) {
-    return ev.key === 'Escape' && this.hide();
+  /**
+   * @param {KeyboardEvent} ev
+   * @private
+   */
+  __escKeyHandler(ev) {
```

Comment:
Maybe we can also check this.handleHidesOnOutsideEsc here (instead of overriding the `__escKeyHandler`in `_handleHidesOnOutsideEsc`...)

(Know you didn't write it, but while we're at it :))

Reply:
Yes, but we don't want to have the `_blocksEscKeyHandler` in that function.
Then its better to have a separate `escKeyOutisideHandler`